doc/guides/collaborator-guide.md

```diff
@@ -1,4 +1,4 @@
-# Node.js Collaborator Guide
+# Node.js collaborator guide
```

Comment:
Should this follow https://docs.microsoft.com/en-us/style-guide/capitalization#title-style-capitalization?

Reply:
I agree. I have made changes accordingly.